src/core/value.jl

```diff
@@ -40,9 +40,10 @@ end
 
 ## general APIs
 
-export llvmtype, name, name!, replace_uses!, isconstant, isundef, context
+export llvmtype, llvmeltype, name, name!, replace_uses!, isconstant, isundef, context
 
 llvmtype(val::Value) = LLVMType(API.LLVMTypeOf(val))
+llvmeltype(val::Value) = LLVMType(API.LLVMGetElementType(llvmtype(x)))
```

Comment:
That should just be `eltype`?
```suggestion
llvmeltype(val::Value) = eltype(llvmtype(x))
```

Reply:
Oh, I should have looked more carefully, I didn't realize that `eltype` worked for this!

Reply:
if you don't want a new export I can just use `eltype` directly